testing/test-setup-config/src/lib/vitest-setup-presets.unit.test.ts

```diff
@@ -0,0 +1,331 @@
+      portalClient: ['testing/test-setup/src/lib/portal-client.mock.ts'],
+      matchersCore: [
+        'testing/test-setup/src/lib/extend/markdown-table.matcher.ts',
+        'testing/test-setup/src/lib/extend/jest-extended.matcher.ts',
+      ],
+      matcherPath: ['testing/test-setup/src/lib/extend/path.matcher.ts'],
+    });
+  });
+
+  it('should export correct integration setup presets', () => {
+    expect(setupPresets.int).toEqual({
+      base: [
+        'testing/test-setup/src/lib/console.mock.ts',
+        'testing/test-setup/src/lib/reset.mocks.ts',
+      ],
+      cliui: ['testing/test-setup/src/lib/cliui.mock.ts'],
+      fs: ['testing/test-setup/src/lib/fs.mock.ts'],
+      git: ['testing/test-setup/src/lib/git.mock.ts'],
+      portalClient: ['testing/test-setup/src/lib/portal-client.mock.ts'],
+      matcherPath: ['testing/test-setup/src/lib/extend/path.matcher.ts'],
+      chromePath: ['testing/test-setup/src/lib/chrome-path.mock.ts'],
+    });
+  });
+
+  it('should export correct e2e setup presets', () => {
+    expect(setupPresets.e2e).toEqual({
+      base: ['testing/test-setup/src/lib/reset.mocks.ts'],
+    });
+  });
+
+  it('should be defined as a const object', () => {
+    expect(setupPresets).toBeDefined();
+    expect(typeof setupPresets).toBe('object');
+  });
+});
+
+// Parameterized tests to eliminate duplication
+describe.each([
+  ['unit', setupPresets.unit.base, createUnitConfig],
+  ['int', setupPresets.int.base, createIntConfig],
+  ['e2e', setupPresets.e2e.base, createE2eConfig],
+] as const)('%s config creation', (kind, baseSetupFiles, createFn) => {
+  beforeEach(() => {
+    vi.clearAllMocks();
+  });
+
+  it('should call createVitestConfig with correct parameters and default setupFiles', () => {
+    createFn(MOCK_PROJECT_KEY, MOCK_CONFIG_REST_PARAMS);
+
+    expect(configFactory.createVitestConfig).toHaveBeenCalledWith(
+      {
+        projectKey: MOCK_PROJECT_KEY,
+        kind,
+        ...MOCK_CONFIG_REST_PARAMS,
+      },
+      {
+        test: {
+          setupFiles: baseSetupFiles,
+        },
+      },
+    );
+  });
+
+  it('should use custom setupFiles from overrides when provided', () => {
+    const customSetupFiles = [`${kind}-setup1.ts`, `${kind}-setup2.ts`];
+    const overrides: VitestOverrides = {
+      test: {
+        setupFiles: customSetupFiles,
+      },
+    };
+
+    createFn(MOCK_PROJECT_KEY, MOCK_CONFIG_REST_PARAMS, overrides);
+
+    expect(configFactory.createVitestConfig).toHaveBeenCalledWith(
+      {
+        projectKey: MOCK_PROJECT_KEY,
+        kind,
+        ...MOCK_CONFIG_REST_PARAMS,
+      },
+      {
+        test: {
+          setupFiles: customSetupFiles,
+        },
+      },
+    );
+  });
+
+  it('should merge other overrides correctly while using default setupFiles', () => {
+    const testTimeout =
+      kind === 'unit'
+        ? TEST_TIMEOUTS.MEDIUM
+        : kind === 'int'
+          ? TEST_TIMEOUTS.LONG
+          : TEST_TIMEOUTS.E2E;
```

Comment:
Let's avoid conditionals in tests. It makes the test code harder to read and adds noise.